main.cpp

```diff
@@ -61,124 +69,411 @@ New/This/Pointers/References conclusion
+    *pointerToFloatValue /= 0;
+    return *this;      
+}
+
+DoubleType& DoubleType::add ( const FloatType& ft )
+{
+    *pointerToDoubleValue += *ft.pointerToFloatValue;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract ( const FloatType& ft ) 
+{ 
+    *pointerToDoubleValue -= *ft.pointerToFloatValue;
+    return *this; 
+}
+
+DoubleType& DoubleType::multiply ( const FloatType& ft ) 
+{ 
+    *pointerToDoubleValue *= *ft.pointerToFloatValue;
+    return *this; 
+}
+
+DoubleType& DoubleType::divide ( const FloatType& ft ) 
+{ 
+    if (abs(*ft.pointerToFloatValue) > 0) 
+    {
+        *pointerToDoubleValue /= *ft.pointerToFloatValue;
+        return *this;    
+    }
+    std::cout << "Divide-by-zero warning  " << std::endl;
+    *pointerToDoubleValue /= 0;
+    return *this;      
+}
+
+DoubleType& DoubleType::add ( const IntType& it )
+{
+    *pointerToDoubleValue += *it.pointerToIntValue;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract ( const IntType& it ) 
+{ 
+    *pointerToDoubleValue -= *it.pointerToIntValue;
+    return *this; 
+}
+
+DoubleType& DoubleType::multiply ( const IntType& it ) 
+{ 
+    *pointerToDoubleValue *= *it.pointerToIntValue;
+    return *this; 
+}
+
+DoubleType& DoubleType::divide ( const IntType& it ) 
+{ 
+    if (abs(*it.pointerToIntValue) > 0) 
+    {
+        *pointerToDoubleValue /= *it.pointerToIntValue;    
+        return *this;
+    }
+    std::cout << "Divide-by-zero warning  " << std::endl;
+    *pointerToDoubleValue /= 0;
+    return *this;      
+}
+
+IntType& IntType::add ( const FloatType& ft )
+{
+    *pointerToIntValue += *ft.pointerToFloatValue;
+    return *this;
+}
+
+IntType& IntType::subtract ( const FloatType& ft ) 
+{ 
+    *pointerToIntValue -= *ft.pointerToFloatValue;
+    return *this;
+}
+
+IntType& IntType::multiply ( const FloatType& ft ) 
+{ 
+    *pointerToIntValue *= *ft.pointerToFloatValue;
+    return *this; 
+}
+
+IntType& IntType::divide ( const FloatType& ft ) 
+{ 
+    if (*ft.pointerToFloatValue > 0.0001f) 
+    {
+        *pointerToIntValue /= *ft.pointerToFloatValue;
+        return *this; 
+    }
+    std::cout << "Cannot Divide Int by Zero! " << std::endl;
+    *pointerToIntValue = 0;
+    return *this;
+}
+
+IntType& IntType::add ( const DoubleType& dt )
+{
+    *pointerToIntValue += *dt.pointerToDoubleValue;
+    return *this;
+}
+
+IntType& IntType::subtract ( const DoubleType& dt ) 
+{ 
+    *pointerToIntValue -= *dt.pointerToDoubleValue;
+    return *this;
+}
+
+IntType& IntType::multiply ( const DoubleType& dt ) 
+{ 
+    *pointerToIntValue *= *dt.pointerToDoubleValue;
+    return *this; 
+}
+
+IntType& IntType::divide ( const DoubleType& dt ) 
+{ 
+    if (*dt.pointerToDoubleValue > 0) 
+    {
+        *pointerToIntValue /= *dt.pointerToDoubleValue; 
+        return *this;
+    }
     std::cout << "Cannot Divide Int by Zero! " << std::endl;
-    return 0;
+    *pointerToIntValue = 0;
+    return *this;
```

Comment:
All of these should just return the result of calling the function that takes the primitive.  
i.e. `return multiply( *dt.pointerToDoubleValue );`

- [ ] D.R.Y. refactor